zmq.hpp

```diff
@@ -436,8 +436,6 @@ class message_t
       class Char,
       size_t N,
       typename = typename std::enable_if<detail::is_char_type<Char>::value>::type>
-    ZMQ_DEPRECATED("from 4.7.0, use constructors taking iterators, (pointer, size) "
```

Comment:
Can you add a guard for C++14 around this instead (I'm assuming the failing version of VS does not support that version), since this function is deprecated should not be used anymore.

Reply:
> I'm assuming the failing version of VS does not support that version

it does!

Reply:
Hm, this is somwhat awkward. Would it work when defining `ZMQ_DEPRECATED` using `declspec` for that MSVC version? e.g. by changing the definition of that macro to
```
#if defined(ZMQ_CPP14) && !defined(_MSC_VER)
#define ZMQ_DEPRECATED(msg) [[deprecated(msg)]]
#elif defined(_MSC_VER)
#define ZMQ_DEPRECATED(msg) __declspec(deprecated(msg))
#elif defined(__GNUC__)
#define ZMQ_DEPRECATED(msg) __attribute__((deprecated(msg)))
#endif
```
and leaving the deprecation mark here unchanged?

Reply:
@gummif Actually, both the VS2017 and VS2015 CI jobs were broken by #399, though I think they were fine on the PR: https://ci.appveyor.com/project/zeromq/cppzmq/builds/35485409

Reply:
Yes could work.

Reply:
Could we remove the deprecation warning which just breaks the library and release 4.7.1, and deprecate the API formally at the next patch release when we have a clean way of handling it?

Reply:
If the suggested fix above work for you then that looks like a good fix to me.